playwright/test/search-works.test.ts

```diff
@@ -1,336 +1,282 @@
-  expectedKey: string,
-  expectedVal: string | undefined,
-  page: Page
-) => {
-  console.info('expectSearchParam', { expectedKey, expectedVal });
-  const params = new URLSearchParams(page.url());
-
-  if (expectedVal) {
-    const foundMatchingParam = Array.from(params).find(([key, val], i) => {
-      if (i === 0) {
-        // The first key returns the whole URL.
-        // This is fine if it has "cachebust" which we ignored,
-        // but if the first value is a valid filter, it never matches.
-        return (
-          key.slice(key.indexOf('?') + 1) === expectedKey && val === expectedVal
-        );
-      } else {
-        return key === expectedKey && val === expectedVal;
-      }
-    });
-
-    if (!foundMatchingParam) {
-      console.log(page.url());
-    }
-
-    expect(foundMatchingParam).toBeTruthy();
-  } else {
-    const noMatchProps = !Array.from(params).find(
-      ([key]) => key === expectedKey
-    );
+test.describe.configure({ mode: 'parallel' });
 
-    expect(noMatchProps).toBeTruthy();
-  }
-};
+test('(1.1) | The user is looking for an archive; it should be browsable from the search results', async ({
+  page,
+  context,
+}) => {
+  await newWorksSearch(context, page);
+
+  // searchFor
+  await page.fill(worksSearchForm, 'Persian');
+  await page.press(worksSearchForm, 'Enter');
 
-const openDropdown = async (label: string, page: Page) => {
-  console.info('openDropdown', label);
+  // selectFilter
   if (isMobile(page)) {
     await page.click(formatFilterMobileButton);
   } else {
-    await page.click(`button :text("${label}")`);
+    await page.click('button[aria-controls="workType"]');
   }
-};
 
-const selectCheckbox = async (label: string, page: Page) => {
-  console.info('selectCheckbox', label);
-  await Promise.all([
-    safeWaitForNavigation(page),
-    page.click(`label :text("${label}")`),
-  ]);
-  await page.waitForTimeout(1000);
+  const filterCheckbox = page.locator(
+    'input[form="search-page-form"][type="checkbox"][name="workType"][value="h"]'
+  ); // Formats > Archives and manuscripts
+
+  await filterCheckbox.click();
+  await expect(filterCheckbox).toBeChecked();
 
   if (isMobile(page)) {
     await page.click(`"Show results"`);
   }
-};
-
-const navigateToNextPage = async (page: Page) => {
-  // data-testid is only set on Pagination components that aren't hidden on mobile
-  // in `common/views/components/Pagination/Pagination.tsx`
-  await page.waitForTimeout(2000);
-
-  await Promise.all([
-    safeWaitForNavigation(page),
-    page.click('[data-testid="pagination"] a'),
-  ]);
-};
-
-const navigateToResult = async (n = 1, page: Page) => {
-  const result = `main ul li:nth-of-type(${n}) a`;
-  await page.waitForTimeout(1000);
+
+  // navigateToNextPage
+  const nextButton = page
+    .locator('[data-testid="pagination"]')
+    .getByRole('link', { name: 'Next' });
+
+  await nextButton.click();
+
+  // confirm navigateToNextPage
+  await expect(page).toHaveURL(/workType=h/);
```

Comment:
Do we need this line? I think that the line below should suffice, and it doesn't rely on knowing what the URL should look like

Reply:
No this will be going!